tests/selenium/test_selenium_chat_flow.py

```
@@ -369,3 +369,238 @@ def test_explain_denial_flow_end_to_end(self):
+            "return localStorage.getItem('fhi_use_external_models');"
+        )
+        assert external_pref == "true", f"External models should be saved, got: {external_pref}"
+
+        # Verify the chat loaded with initial message (denial text should appear)
+        page_content = self.get_page_source()
+        assert "MRI" in page_content or "denied" in page_content.lower(), \
+            "Chat should show the denial message"
+
+        print("✓ Explain denial with external models works correctly")
+        print(f"  - External models preference saved: {external_pref}")
+        print(f"  - Initial message sent to chat")
```

Ruff (0.14.10)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)